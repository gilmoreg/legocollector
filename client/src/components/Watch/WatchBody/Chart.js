import React from 'react';
import PropTypes from 'prop-types';
// https://github.com/jerairrest/react-chartjs-2/blob/master/example/src/components/line.js
import { Line } from 'react-chartjs-2';
import moment from 'moment';
import './Chart.css';

const Chart = (props) => {
  const data = {
    labels: props.stock_levels.map(level =>
      moment(new Date(level.datetime)).format('MM-DD')),
    datasets: [
      {
        label: false,
        fill: false,
        lineTension: 0.1,
        backgroundColor: 'rgba(75,192,192,0.4)',
        borderColor: 'rgba(75,192,192,1)',
        borderCapStyle: 'butt',
        borderDash: [],
        borderDashOffset: 0.0,
        borderJoinStyle: 'miter',
        pointBorderColor: 'rgba(75,192,192,1)',
        pointBackgroundColor: '#fff',
        pointBorderWidth: 1,
        pointHoverRadius: 5,
        pointHoverBackgroundColor: 'rgba(75,192,192,1)',
        pointHoverBorderColor: 'rgba(220,220,220,1)',
        pointHoverBorderWidth: 2,
        pointRadius: 1,
        pointHitRadius: 10,
        data: props.stock_levels.map(level => level.stock_level),
      },
    ],
  };
  const options = {
    legend: {
      display: false,
    },
  };

  return (
    <div className="Chart">
      <Line data={data} options={options} />
    </div>
  );
};

Chart.propTypes = {
  stock_levels: PropTypes.arrayOf(PropTypes.shape({
    datetime: PropTypes.string,
    id: PropTypes.number,
    stock_level: PropTypes.number,
  })).isRequired,
};

export default Chart;
