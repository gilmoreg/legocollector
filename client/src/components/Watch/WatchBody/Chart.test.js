import React from 'react';
import { shallow } from 'enzyme';
import Chart from './Chart';

it('renders without crashing', () => {
  const stockLevels = [
    { datetime: 'test', stock_level: 1 },
    { datetime: 'test', stock_level: 1 },
  ];
  const wrapper = shallow(
    <Chart stock_levels={stockLevels} />,
  );
  expect(wrapper).toMatchSnapshot();
});
