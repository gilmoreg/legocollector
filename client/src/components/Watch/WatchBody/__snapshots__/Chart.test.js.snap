// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Chart
    stock_levels={
      Array [
        Object {
          "datetime": "test",
          "stock_level": 1,
        },
        Object {
          "datetime": "test",
          "stock_level": 1,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <Line
        data={
          Object {
            "datasets": Array [
              Object {
                "backgroundColor": "rgba(75,192,192,0.4)",
                "borderCapStyle": "butt",
                "borderDash": Array [],
                "borderDashOffset": 0,
                "borderJoinStyle": "miter",
                "data": Array [
                  1,
                  1,
                ],
                "fill": false,
                "label": false,
                "lineTension": 0.1,
                "maintainAspectRatio": false,
                "pointBackgroundColor": "#fff",
                "pointBorderColor": "rgba(75,192,192,1)",
                "pointBorderWidth": 1,
                "pointHitRadius": 10,
                "pointHoverBackgroundColor": "rgba(75,192,192,1)",
                "pointHoverBorderColor": "rgba(220,220,220,1)",
                "pointHoverBorderWidth": 2,
                "pointHoverRadius": 5,
                "pointRadius": 1,
                "responsive": true,
              },
            ],
            "labels": Array [
              "Invalid date",
              "Invalid date",
            ],
          }
        }
        height={200}
        options={
          Object {
            "legend": Object {
              "display": false,
            },
            "scales": Object {
              "yAxes": Array [
                Object {
                  "display": false,
                  "gridLines": Object {
                    "drawBorder": false,
                  },
                },
              ],
            },
          }
        }
        width={300}
      />,
      "className": "Chart",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "data": Object {
          "datasets": Array [
            Object {
              "backgroundColor": "rgba(75,192,192,0.4)",
              "borderCapStyle": "butt",
              "borderDash": Array [],
              "borderDashOffset": 0,
              "borderJoinStyle": "miter",
              "data": Array [
                1,
                1,
              ],
              "fill": false,
              "label": false,
              "lineTension": 0.1,
              "maintainAspectRatio": false,
              "pointBackgroundColor": "#fff",
              "pointBorderColor": "rgba(75,192,192,1)",
              "pointBorderWidth": 1,
              "pointHitRadius": 10,
              "pointHoverBackgroundColor": "rgba(75,192,192,1)",
              "pointHoverBorderColor": "rgba(220,220,220,1)",
              "pointHoverBorderWidth": 2,
              "pointHoverRadius": 5,
              "pointRadius": 1,
              "responsive": true,
            },
          ],
          "labels": Array [
            "Invalid date",
            "Invalid date",
          ],
        },
        "height": 200,
        "options": Object {
          "legend": Object {
            "display": false,
          },
          "scales": Object {
            "yAxes": Array [
              Object {
                "display": false,
                "gridLines": Object {
                  "drawBorder": false,
                },
              },
            ],
          },
        },
        "width": 300,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <Line
          data={
            Object {
              "datasets": Array [
                Object {
                  "backgroundColor": "rgba(75,192,192,0.4)",
                  "borderCapStyle": "butt",
                  "borderDash": Array [],
                  "borderDashOffset": 0,
                  "borderJoinStyle": "miter",
                  "data": Array [
                    1,
                    1,
                  ],
                  "fill": false,
                  "label": false,
                  "lineTension": 0.1,
                  "maintainAspectRatio": false,
                  "pointBackgroundColor": "#fff",
                  "pointBorderColor": "rgba(75,192,192,1)",
                  "pointBorderWidth": 1,
                  "pointHitRadius": 10,
                  "pointHoverBackgroundColor": "rgba(75,192,192,1)",
                  "pointHoverBorderColor": "rgba(220,220,220,1)",
                  "pointHoverBorderWidth": 2,
                  "pointHoverRadius": 5,
                  "pointRadius": 1,
                  "responsive": true,
                },
              ],
              "labels": Array [
                "Invalid date",
                "Invalid date",
              ],
            }
          }
          height={200}
          options={
            Object {
              "legend": Object {
                "display": false,
              },
              "scales": Object {
                "yAxes": Array [
                  Object {
                    "display": false,
                    "gridLines": Object {
                      "drawBorder": false,
                    },
                  },
                ],
              },
            }
          }
          width={300}
        />,
        "className": "Chart",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "data": Object {
            "datasets": Array [
              Object {
                "backgroundColor": "rgba(75,192,192,0.4)",
                "borderCapStyle": "butt",
                "borderDash": Array [],
                "borderDashOffset": 0,
                "borderJoinStyle": "miter",
                "data": Array [
                  1,
                  1,
                ],
                "fill": false,
                "label": false,
                "lineTension": 0.1,
                "maintainAspectRatio": false,
                "pointBackgroundColor": "#fff",
                "pointBorderColor": "rgba(75,192,192,1)",
                "pointBorderWidth": 1,
                "pointHitRadius": 10,
                "pointHoverBackgroundColor": "rgba(75,192,192,1)",
                "pointHoverBorderColor": "rgba(220,220,220,1)",
                "pointHoverBorderWidth": 2,
                "pointHoverRadius": 5,
                "pointRadius": 1,
                "responsive": true,
              },
            ],
            "labels": Array [
              "Invalid date",
              "Invalid date",
            ],
          },
          "height": 200,
          "options": Object {
            "legend": Object {
              "display": false,
            },
            "scales": Object {
              "yAxes": Array [
                Object {
                  "display": false,
                  "gridLines": Object {
                    "drawBorder": false,
                  },
                },
              ],
            },
          },
          "width": 300,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
